main.c

```diff
@@ -3910,6 +3925,15 @@ ovl_setattr (fuse_req_t req, fuse_ino_t ino, struct stat *attr, int to_set, stru
           strconcat3 (path, PATH_MAX, get_upper_layer (lo)->path, "/", node->path);
           break;
         }
+      if (need_restore_mode)
+        {
+          ret = fchmodat(dirfd, node->path, node->ino->mode, 0);
```

Comment:
it doesn't restore the mode to the original one, we cannot assume the original mode was `0`

Reply:
The mode to be set by `fchmodat` here is `node->ino->mode` rather than `0`.
c.f. 
```c
int fchmodat(int dirfd, const char *pathname, mode_t mode, int flags);
```